Scripts/Runtime/Collections/UnsafeBuffer.cs

```diff
@@ -0,0 +1,550 @@
+            internal LaneReader(Lane* lane, BufferInfo* bufferInfo)
+            {
+                m_Lane = lane;
+                m_BufferInfo = bufferInfo;
+                
+                m_CurrentReadBlock = m_Lane->FirstBlock;
+                m_ReaderHead = (m_CurrentReadBlock != null) 
+                    ? m_CurrentReadBlock->Data
+                    : null;
+                m_EndOfCurrentBlock = m_ReaderHead + m_BufferInfo->BlockSize;
+            }
+
+            private void CheckForEndOfBlock()
+            {
+                if (m_ReaderHead < m_EndOfCurrentBlock)
+                {
+                    return;
+                }
+                
+                m_CurrentReadBlock = m_CurrentReadBlock->Next;
+                m_ReaderHead = (m_CurrentReadBlock != null) 
+                    ? m_CurrentReadBlock->Data
+                    : null;
+                m_EndOfCurrentBlock = m_ReaderHead + m_BufferInfo->BlockSize;
+            }
+            
+            /// <summary>
+            /// Reads the next element from the reader
+            /// </summary>
+            /// <returns>The next element</returns>
+            public ref T Read()
+            {
+                //Otherwise nothing has been written
+                //TODO: Write test for this
+                Assert.IsTrue(Count > 0 && m_CurrentReadBlock != null && m_ReaderHead != null);
+
+                byte* readPointer = m_ReaderHead;
+                m_ReaderHead += m_BufferInfo->ElementSize;
+                
+                CheckForEndOfBlock();
+                
+                return ref UnsafeUtility.AsRef<T>(readPointer);
+            }
+            
+            /// <summary>
+            /// Peeks at what the next elements in the reader is but does not advance the reader head.
+            /// Will return the same value until <see cref="Read"/> has been called.
+            /// </summary>
+            /// <returns>The next element</returns>
+            public ref T Peek()
+            {
+                ///Otherwise nothing has been written
+                //TODO: Write test for this
+                Assert.IsTrue(Count > 0 && m_CurrentReadBlock != null && m_ReaderHead != null);
+                
+                return ref UnsafeUtility.AsRef<T>(m_ReaderHead);
+            }
+        }
+        
+        //*************************************************************************************************************
+        // JOBS
+        //*************************************************************************************************************
+        
+        [BurstCompile]
+        private struct DisposeJob : IJob
+        {
+            private UnsafeBuffer<T> m_UnsafeBuffer;
+
+            public DisposeJob(UnsafeBuffer<T> unsafeBuffer)
+            {
+                m_UnsafeBuffer = unsafeBuffer;
+            }
+
+            public void Execute()
+            {
+                m_UnsafeBuffer.Dispose();
+            }
+        }
```

Comment:
Maybe a good case for a generic `DisposeJob` that accepts `IDisposable`?

I actually thought we had something like that. I may have created that job at some point and deleted it when it wasn't needed anymore.

Reply:
I suppose we could, I just followed the convention for all the other collections that exist right now.